### ISSUE TITLE
Creating a build / release pipeline pointing to a YAML file

### ISSUE THREAD

**Issue by juanperezADD** [microsoft/azure-devops-dotnet-samples#251]
We are using Azure DevOps Server 2019 and we want to import our existing build / release pipelines which are created based on a YAML. 

In this post, https://developercommunity.visualstudio.com/content/problem/428975/azure-devops-rest-api-to-create-a-build-definition.html, says that we can not do this with REST API, it is only supported with the Visual Designer. But, can we do it with the .NET Client library?

Thank you in advance

**Comment by lurumad** [comment]
Any news about this? @danhellem @willsmythe @bansalaseem

We need to use YAML pipelines from .NET libraries.

Regards!

**Comment by wnjenkin** [comment]
This is supported for classic pipelines instead of YAML pipelines. We don't have any api which supports import for YAML.

**Comment by juanperezADD** [comment]
Do you know if there is any plan to support this in the .NET library or the REST API?

**Comment by wnjenkin** [comment]
We've talked about adding support for this scenario, however we don't have any ETA on when this functionality would be delivered.

**Comment by juanperezADD** [comment]
This feature is really important for us because we are looking for a zero touch YAML build creation. 

We would really appreciate if this feature climbs on the backlog list.

**Comment by lurumad** [comment]
Same fors us.

**Comment by willsmythe** [comment]
In the meantime, you can use the [Azure DevOps CLI](https://docs.microsoft.com/en-us/azure/devops/cli/get-started?view=azure-devops) to create a YAML-based Pipeline. 

See the [`az pipelines create`](https://docs.microsoft.com/en-us/cli/azure/ext/azure-devops/pipelines?view=azure-cli-latest#ext-azure-devops-az-pipelines-create) command. Here's an example:

```bash
az pipelines create --name 'ContosoBuild' --description 'Pipeline for contoso project'
--repository https://github.com/SampleOrg/SampleRepo --branch master --yml-path azure-pipelines.yml
```

**Comment by juanperezADD** [comment]
Thanks for your answer but we cannot use [Azure DevOps CLI](https://docs.microsoft.com/en-us/azure/devops/cli/get-started?view=azure-devops&viewFallbackFrom=azure-devops-2019) because it is not available in Azure DevOps Server 2019.

**Comment by willsmythe** [comment]
Yep, that's right (sorry for not reading your description carefully and/or assuming cloud). For others needing to create YAML pipelines on Azure DevOps Services, `az pipelines create` is your best option for now ...

**Comment by juanperezADD** [comment]
Where can we create a feature request to be taken into account in the backlog?

This [one ](https://developercommunity.visualstudio.com/content/problem/628532/import-builds-based-on-yaml-through-the-rest-api-o-1.html?childToView=750010#comment-750010)was recently closed...

Sorry for the insistence but it is very important for us.

**Comment by wnjenkin** [comment]
You can create a new feature request using the following url. Before creating a new request, please check to make sure there isn't already a similar request. If there is a similar request, you can upvote to boost priority.

Create new feature request - https://developercommunity.visualstudio.com/content/idea/post.html?space=21

**Comment by Tenseiga** [comment]
seriously guys, Can I know how the current ui, does that without exposing any of APIs ?
